client/ayon_houdini/plugins/publish/extract_rop.py

```diff
@@ -33,9 +34,31 @@ def process(self, instance: pyblish.api.Instance):
         )
         ext = ext.lstrip(".")
 
-        self.log.debug(f"Rendering {rop_node.path()} to {first_file}..")
-
-        render_rop(rop_node)
+        self.log.debug(
+            "Rendering {node_path} to {location}".format(
+                node_path=rop_node.path(),
+                location=instance.data['stagingDir'] if isinstance(files, (list, tuple)) else files
+            )
+        )
+        frames_to_fix = clique.parse(instance.data.get("frames_to_fix", ""), "{ranges}")
+
+        if len(set(frames_to_fix)) > 1:
+            # Render only frames to fix
+            for frame_range in frames_to_fix.separate():
+                frame_range = list(frame_range)
+                self.log.debug(
+                    "Rendering frames to fix [{f1}, {f2}]".format(
+                        f1=frame_range[0],
+                        f2=frame_range[-1]
+                    )
+                )
+                frame_range = (
+                    int(frame_range[0]), int(frame_range[-1]), instance.data["byFrameStep"]
+                )
+                render_rop(rop_node, frame_range=frame_range)
```

Comment:
This logic feels incorrect. If we do a `byFrameStep` for each separate chunk. Aren't we stepping from the wrong start frame and hence offsetted in each chunk.

Also, we may as well off load this to the `render_rop` function so we don't need to duplicate the logic.

So that we just do `render_rop(rop_nopes, frames_to_render)`?

Reply:
That's a good idea.  but, it'll be frames to fix which is a string containing the frame range `"1001-1003"`.
`render_rop(rop_nopes, frames_to_fix)` .. I may need some time to think about it. because that string doesn't contain any info about the frame step.

> This logic feels incorrect. If we do a `byFrameStep` for each separate chunk. Aren't we stepping from the wrong start frame and hence offsetted in each chunk.

I missed this 😢 Any ideas about considering this ?

Reply:
The solution to this @MustafaJafar - is whenever you specify `frames_to_fix` to FORCE `byFrameStep` to 1.0. If you still want to "step frames" for frames to fix, you should then do, e.g. `1-100x5` (does clique parse support that?)

Reply:
> `1-100x5` (does clique parse support that?)

No.

Reply:
https://github.com/ynput/ayon-houdini/pull/45/commits/6be6494e0ccca718e21c839d5d380d403d32dceb